src/reducers/scrollState.js

```diff
@@ -72,14 +71,18 @@ function scrollState(state = DEFAULT_STATE, action) {
 
       if (newProps.scrollTop && newProps.scrollTop !== oldProps.scrollTop) {
         state = scrollTo(state, newProps.scrollTop);
-        state = updateVisibleRows(state);
       }
 
       if (newProps.scrollToRow && newProps.scrollToRow !== oldProps.scrollToRow) {
         state = scrollToRow(state, newProps.scrollToRow);
-        state = updateVisibleRows(state);
       }
 
+      if (newProps.rowsCount !== state.rowsCount) {
+        state = updateRowCount(state, newProps);
+      }
+
+      state = updateVisibleRows(state);
```

Comment:
Yeah that should work, and Kam's correct this was definitely harmful to performance. Do we have any tests in place to watch for performance regressions? I didn't notice anything like that..